### DISCUSSION TITLE
[Feature request] Extend top/left/bottom/right to support 50%,100%

### DISCUSSION THREAD

**Discussion by misha-erm** [tailwindlabs/tailwindcss#1396] [Ideas]
Hello,

I'd like to know you thoughts about adding support for positioning styles like:
```
top/left/bottom/right: 100%/50%
```
It can be useful together with absolutely positioned elements. E.g. centering element (`top: 50%; transform: translateY(-50%)` )

Thanks in advance

**Comment by tlgreg** [comment]
You can.
https://tailwindcss.com/docs/top-right-bottom-left/#customizing

**Comment by misha-erm** [comment]
thanks. 
But thought of adding it to the core by default since we have utilities like translate-x-1/2

**Comment by adamwathan** [comment]
Agree these are useful to have by default, adding soon 👍🏻

**Comment by garrettmaring** [comment]
I've found myself doing this. Might want to add this to the default as well.

```js
extend: {
  inset: (theme, { negative }) => ({ ...theme('spacing'), ...negative(theme('spacing')) }),
},
```

**Comment by magicspon** [comment]
I go even further than @garrettmaring, i add all of the width values to inset as well. 

left-3/12, top-1/2 etc etc...

**Comment by tlgreg** [comment]
While the config can be extended with a lot of useful values and most of us use purgecss to keep what's actually used but the defaults should consider the CDN build too.

**Comment by rightaway** [comment]
Is it possible to `extend` the inset values or the only option is to override them? I know there's only 2 values for inset, but can I still reference the existing values so I can add to them?

**Comment by tlgreg** [comment]
Yes, you can, generally `extend` values of the `theme`.
Docs:
<https://tailwindcss.com/docs/theme/#extending-the-default-theme>

In this case it would look like:
```js
module.exports = {
  theme: {
    extend: {
      inset: {
        '1/2': '50%',
        full: '100%',
      },
    },
  },
}
```

**Comment by gremo** [comment]
I agree that these utilities should be added by default. It's very common to center vertically with `absolute top-1/2 transform -translate-y-1/2` or horizontally with  `absolute left-1/2 transform -translate-x-1/2`.

**Reply by FullStackJohnMack** [reply]
bump for this

**Reply by adamwathan** [reply]
These exist by default as of v2.0 which came out in November 👍🏻
